metabase/metabase
# Migrate new dashboard form to formik

Epic #26178

Migrates the `CreateDashboardModal` component to `formik`. A few changes along the way:

* breaks `CreateDashboardModal` into `CreateDashboardForm` and `CreateDashboardModal` components
* adds shared `FormFooter` component
* adds shared `FormCollectionPicker` component
* tweaks `FormProvider's` template types, so forms can be easily typed

### To Verify

1. Click "New" at the top right
2. Pick "Dashboard"
3. Ensure you can't submit the form until the name is filled in
4. Ensure the dashboard is actually saved
5. Try with/without a description; try different collections

### Demo


https://user-images.githubusercontent.com/17258145/204334497-192110ea-2a55-42a3-8e55-834200f3840d.mp4
